usr/libexec/greenboot/greenboot-grub2-set-counter

```diff
@@ -14,7 +14,31 @@ elif [ -n "$GREENBOOT_MAX_BOOT_ATTEMPTS" ]; then
 else
     max_boot_attempts=3 # default to 3 attempts
 fi
-grub2-editenv - set boot_counter="$max_boot_attempts"
-grub2-editenv - set boot_success=0
+
+# Remount /boot as read-only if it was mounted as read-write
+if grep -q " /boot .* rw," /proc/mounts; then
+    mount -o remount,ro /boot || exit 1
+fi
```

Comment:
You are doing the same check of `/boot` and remounting it multiple times in the code...maybe an opportunity to create a reusable function?

Reply:
moved the ro to a sperate function to make code cleaner